src/transformers/models/colpali/modular_colpali.py

```diff
@@ -0,0 +1,336 @@
+    ColPaliForRetrieval according to the specified arguments, defining the model architecture.
+
+    The ColPali config is stricly equivalent to the PaliGemma config, but with a different model type.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.model_type = "colpali"
+        self.is_composition = False
+
+
+class ColPaliProcessor(PaliGemmaProcessor):
+    r"""
+    Processor for ColPali.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mock_image = Image.new("RGB", (16, 16), color="black")
+
+    @staticmethod
+    def get_torch_device(device: str = "auto") -> str:
+        """
+        Returns the device (string) to be used by PyTorch.
+
+        `device` arg defaults to "auto" which will use:
+        - "cuda:0" if available
+        - else "mps" if available
+        - else "cpu".
+        """
+
+        if device == "auto":
+            if torch.cuda.is_available():
+                device = "cuda:0"
+            elif torch.backends.mps.is_available():  # for Apple Silicon
+                device = "mps"
+            else:
+                device = "cpu"
+            logger.info(f"Using device: {device}")
+
+        return device
+
+    def process_images(
+        self,
+        images: List[Image.Image],
+    ) -> BatchFeature:
+        """
+        Process images for ColPali.
+        """
+        texts_doc = ["Describe the image."] * len(images)
+        images = [image.convert("RGB") for image in images]
+
+        batch_doc = self(
+            text=texts_doc,
+            images=images,
+            return_tensors="pt",
+            padding="longest",
+        )
+        return batch_doc
+
+    def process_queries(
+        self,
+        queries: List[str],
+        max_length: int = 50,
+        suffix: Optional[str] = None,
+    ) -> BatchFeature:
+        """
+        Process queries for ColPali.
+        """
+        if suffix is None:
+            suffix = "<pad>" * 10
+        texts_query: List[str] = []
+
+        for query in queries:
+            query = f"Question: {query}"
+            query += suffix  # add suffix (pad tokens)
+            texts_query.append(query)
+
+        batch_query = self(
+            images=[self.mock_image] * len(texts_query),
+            text=texts_query,
+            return_tensors="pt",
+            padding="longest",
+            max_length=max_length + self.image_seq_length,
+        )
+
+        del batch_query["pixel_values"]
+
+        batch_query["input_ids"] = batch_query["input_ids"][..., self.image_seq_length :]
+        batch_query["attention_mask"] = batch_query["attention_mask"][..., self.image_seq_length :]
+
+        return batch_query
+
+    def score(
```

Comment:
Is this a post processing function to get the retrieval scores? Maybe we could rename it to something like `post_process_retrieval` to follow the formatting for other downstream tasks like object detection